lib/bundlex/toolchain/common/unix/os_deps.ex

```diff
@@ -26,144 +42,171 @@ defmodule Bundlex.Toolchain.Common.Unix.OSDeps do
     }
   end
 
-  defp resolve_single_os_dep([], lib_names, _app) do
-    raise "Couldn't load OS dependencies for libraries: #{inspect(lib_names)}."
-  end
-
-  defp resolve_single_os_dep(providers, lib_names, app) do
-    [first_provider | rest_of_providers] = providers
-
-    case first_provider do
-      nil ->
-        resolve_single_os_dep(rest_of_providers, lib_names, app)
+  defp resolve_os_dep(name, app, [], errors) do
```

Comment:
How about handling separately a case where the errors list is empty (meaning, that no providers were tried)?

Reply:
done